elastic/kibana
# [Upgrade Assistant] Add checkpoint feature to Overview page

### Summary

This PR adds the Checkpoint feature that was left out of https://github.com/elastic/kibana/pull/106521 into the Overview page from Upgrade Assistant.

I left out tests for the LocalStorage persistence part since it's prone to change to a `kibana saved object` soon enough.

**How to test**
1. To enable UA, set `xpack.upgrade_assistant.readonly: false` in your `kibana.yml`
2. Start elasticsearch with `yarn es snapshot --license=trial` 
3. Start kibana with `yarn start`
4. Navigate to UA and enable the toggle to start collecting logs
5. Try navigating to the console and maybe running some commands, that should trigger a few warnings


**Screenshots**

<details>
<summary>Overview of changes</summary>

// Callout with warnings
![Screenshot 2021-08-23 at 15 15 09](https://user-images.githubusercontent.com/1191206/130454993-99ec4448-0cb4-4fc5-ba96-7bf184db889b.png)

// Callout without warnings
![Screenshot 2021-08-23 at 15 24 58](https://user-images.githubusercontent.com/1191206/130455082-bd19afeb-cc57-4f39-a847-b3fb6ed09095.png)

// Loading state
![Screenshot 2021-08-24 at 09 12 36](https://user-images.githubusercontent.com/1191206/130573214-be4ddfc7-2fa5-411b-8cbf-ba53ffd72fb9.png)

// Error state
![Screenshot 2021-08-24 at 09 12 22](https://user-images.githubusercontent.com/1191206/130573243-0bd9fa5e-5aa9-419d-8286-3385cce4ef5e.png)


</details>
